pkg/internal/itest/http_fetch_test.go

```diff
@@ -765,7 +769,9 @@ func TestHttpFetch(t *testing.T) {
 					_, err := uuid.Parse(requestId)
 					req.NoError(err)
 					body, err := io.ReadAll(resp.Body)
-					req.NoError(err)
+					if !testCase.expectUncleanEnd {
+						req.NoError(err)
+					}
```

Comment:
can we get the opposite check: that there is an error if we expect an unclean end?